osu.Game/Skinning/Components/BeatmapAttributeText.cs

```diff
@@ -74,34 +94,32 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
+            ruleset = game.Ruleset.GetBoundCopy();
+
             Attribute.BindValueChanged(_ => updateLabel());
             Template.BindValueChanged(_ => updateLabel());
-            beatmap.BindValueChanged(b =>
+
+            ruleset.BindValueChanged(_ => updateAllContent());
+            beatmap.BindValueChanged(_ =>
             {
-                updateBeatmapContent(b.NewValue);
-                updateLabel();
+                updateBindableDifficulty();
+                updateAllContent();
```

Comment:
Dunno how to feel about the fact that any change to beatmap will do however many diffcalcs and such even if the `BeatmapAttributeText` doesn't even use the value. Or that this is done locally in each instance, which means multiply the overhead times count of `BeatmapAttributeText`s on screen.

The old implementation also did this yes but at least it was _fast_ enough to not matter. This might not be anymore.

Reply:
The only slow calculation here is star rating and it's not calculated but instead taken from cache
As it's uses the same star rating as other displays like map display or display in mod select
But if this bothers you - I will fix this